File: Tyche.Android/data/pool/src/main/java/com/felipearpa/data/pool/domain/PoolGamblerScoreResponse.kt
package com.felipearpa.data.pool.domain

internal data class PoolGamblerScoreResponse(
    val poolId: String,
    val poolName: String,
    val gamblerId: String,
    val gamblerUsername: String,
    val currentPosition: Int?,
    val beforePosition: Int?,
    val score: Int?
)

File: Tyche.iOS/DataPool/Sources/DataPool/Domain/PoolGamblerScoreResponse.swift
struct PoolGamblerScoreResponse : Codable {
    let poolId: String
    let poolName: String
    let gamblerId: String
    let gamblerUsername: String
    let currentPosition: Int?
    let beforePosition: Int?
    let score: Int?
}

extension PoolGamblerScoreResponse {
    func toPoolGamblerScore() -> PoolGamblerScore {
        return PoolGamblerScore(
            poolId: self.poolId,
            poolName: self.poolName,
            gamblerId: self.gamblerId,
            gamblerUsername: self.gamblerUsername,
            currentPosition: self.currentPosition,
            beforePosition: self.beforePosition,
            score: self.score
        )
    }
}
